src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -247,16 +280,37 @@ protected void AddSecrets(TransactionBuildContext context)
         }
 
         /// <summary>
-        /// Find the next available change address.
+        /// If a ChangeAddress was set in the options, check it is from the wallet given and use that for the transaction.
+        /// Otherwise find the next available change address and use that for the transaction.
         /// </summary>
         /// <param name="context">The context associated with the current transaction being built.</param>
         protected void FindChangeAddress(TransactionBuildContext context)
         {
-            // Get an address to send the change to.
-            context.ChangeAddress = this.walletManager.GetUnusedChangeAddress(new WalletAccountReference(context.AccountReference.WalletName, context.AccountReference.AccountName));
+            if (context.BuildOptions.ChangeAddress != null)
```

Comment:
There is a change of functionality here.
Can we maybe add this in another PR?
I'm already not confident I can review this properly.